instrumentation/opentelemetry-instrumentation-falcon/src/opentelemetry/instrumentation/falcon/__init__.py

```diff
@@ -379,12 +411,36 @@ def _start_response(status, response_headers, *args, **kwargs):
             exception = exc
             raise
         finally:
-            if span.is_recording():
-                duration_attrs[SpanAttributes.HTTP_STATUS_CODE] = (
-                    span.attributes.get(SpanAttributes.HTTP_STATUS_CODE)
+            duration_s = default_timer() - start
+            if self.duration_histogram_old:
+                duration_attrs = otel_wsgi._parse_duration_attrs(
+                    attributes, _HTTPStabilityMode.DEFAULT
                 )
-            duration = max(round((default_timer() - start) * 1000), 0)
-            self.duration_histogram.record(duration, duration_attrs)
+                if span.is_recording():
```

Comment:
Thanks for the work so far. I think we want status_code independent of tracing decisions—also, a test showing that a nonrecording span would produce metrics with the correct HTTP status code.

Reply:
On a second thought, I don't think we should set the status code for duration metrics at all, at least I don't see that we do in the other instrumentation wrappers. What are your thoughts on this?

Reply:
The attribute is conditionally required if there is a send/received status code. We are doing exactly the same thing for asgi

Reply:
I was able to solve this by adding the request attributes in the env. Now the status code is properly added to the metric attributes.